addons/crm/static/tests/forecast_kanban_tests.js

```diff
@@ -0,0 +1,262 @@
+            archs: {
+                "lead,false,search": `
+                    <search>
+                        <filter name="forecast" string="Forecast" context="{'forecast_filter':1}"/>
+                        <filter name='groupby_date' context="{'group_by':'date'}"/>
+                        <filter name='groupby_datetime' context="{'group_by':'datetime'}"/>
+                    </search>`
+            },
+            data: {
+                lead: {
+                    fields: {
+                        name: {string: 'Name', type: 'char'},
+                        date: {string: "Date Field", type: 'date'},
+                        datetime: {string: "Datetime Field", type: 'datetime'},
+                    },
+                    records: [
+                        {id: 1, name: 'Lead 1', date: '2021-01-01', datetime: '2021-01-01 00:00:00'},
+                        {id: 2, name: 'Lead 2', date: '2021-01-20', datetime: '2021-01-20 00:00:00'},
+                        {id: 3, name: 'Lead 3', date: '2021-02-01', datetime: '2021-02-01 00:00:00'},
+                        {id: 4, name: 'Lead 4', date: '2021-02-20', datetime: '2021-02-20 00:00:00'},
+                        {id: 5, name: 'Lead 5', date: '2021-03-01', datetime: '2021-03-01 00:00:00'},
+                        {id: 6, name: 'Lead 6', date: '2021-03-20', datetime: '2021-03-20 00:00:00'},
+                    ],
+                },
+            },
+            services: { 'fillTemporalService': FillTemporalService },
+            model: 'lead',
+            View: ForecastKanbanView,
+            viewOptions: {
+                context: {
+                    search_default_forecast: true,
+                    search_default_groupby_date: true,
+                    forecast_field: 'date',
+                },
+            },
+            groupBy: ['date'],
+        };
+        this.unPatchDate = await testUtils.mock.patchDate(2021, 1, 10, 0, 0, 0);
+    },
+    afterEach: async function () {
+        await this.unPatchDate();
+    },
+
+}, function () {
+    QUnit.test("filter out every records before the start of the current month with forecast_filter", async function (assert) {
+        // the filter is used by the forecast model extension, and applies the forecast_filter context key,
+        // which adds a domain constraint on the field marked in the other context key forecast_field
+        assert.expect(14);
+
+        var kanban = await testUtils.createView(this.testKanbanView);
+
+        // the filter is active
+        assert.strictEqual(kanban.$('.o_kanban_group').length, 2, "There should be 2 columns");
+        assert.strictEqual(kanban.$('.o_kanban_group:nth-child(1) .o_kanban_record').length, 2,
+                        "1st column February should contain 2 record");
+        assert.strictEqual(kanban.$('.o_kanban_group:nth-child(2) .o_kanban_record').length, 2,
+                        "2nd column March should contain 2 records");
+        kanban.destroy();
```

Comment:
Make one test for each test case ; when you re-use the same variable for storing a mock view/action this is generally a sign that you should split your test.